src/main.rs

```diff
@@ -18,10 +18,26 @@ use stratus::infra::postgres::Postgres;
 
 fn main() -> anyhow::Result<()> {
     let runtime = new_tokio_runtime("tokio-main", 1, 1); // TODO: make this value configurable
-    runtime.block_on(run_application())
+    runtime.block_on(async {
+        let rpc_handle = tokio::spawn(async {
+            if let Err(e) = run_rpc_server().await {
```

Comment:
I think you need to return the error, otherwise `try_join!` will not know the task failed and will not abort immediately.